src/components/xy_chart/chart.js

```diff
@@ -61,11 +61,19 @@ export class XYChart extends PureComponent {
       return seriesData.x === xBucket
     })[0])
 
-    if(chartDataForXValue.length === 0) {
-      chartDataForXValue.push({ x: xBucket, y: NO_DATA_VALUE })
+    if (chartDataForXValue.length === 0) {
+      chartDataForXValue.push({ x: xBucket, y: NO_DATA_VALUE });
     }
 
-    return chartDataForXValue;
+    return chartDataForXValue
```

Comment:
This is a very expensive thing to do as a mouse moves. Would love to discuss why this is needed here